todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskViewModel.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2016, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.architecture.blueprints.todoapp.addedittask;
+
+import android.content.Context;
+import android.databinding.BaseObservable;
+import android.databinding.ObservableBoolean;
+import android.databinding.ObservableField;
+
+import com.android.annotations.Nullable;
+import com.example.android.architecture.blueprints.todoapp.R;
+import com.example.android.architecture.blueprints.todoapp.SnackBarChangedCallback;
+import com.example.android.architecture.blueprints.todoapp.data.Task;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksDataSource;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
+
+/**
+ * ViewModel for the {@link AddEditTaskFragment}.
+ */
+public class AddEditTaskViewModel extends BaseObservable implements
+        TasksDataSource.GetTaskCallback, SnackBarChangedCallback.SnackBarViewModel {
+
+    public final String TAG = "AddEditTaskViewModel";
+
+    private final TasksRepository mTasksRepository;
```

Comment:
would be nice to have an order of members:
first public then private members. would make things easier to read. but, your choice.